packages/gatsby-plugin-fastify/src/utils/routes.ts

```diff
@@ -12,6 +12,7 @@ export function removeQueryParmsFromUrl(url: string) {
 
 export function buildUrlFromParams(path: string, data: { [s: string]: string } = {}) {
   return path.replace(/:(\w+)|(\*)/gi, function (_match, p1, p2) {
+    if (p1 && !data[p1]) return _match; // :Something in toPath does not have a splat in fromPath pass it through colon intact
```

Comment:
```suggestion
  return path.replace(/:(\w+)|(\*)/gi, function (match, p1, p2) {
    if (p1 && !data[p1]) return match; // :Something in toPath does not have a splat in fromPath pass it through colon intact
```

FYI, the `_` is a convention folks use to indicate an unused variable and silence TSC/ESLint for "unused vars". You can remove that since we're using this now.